### ISSUE TITLE
Vehicle Store Explode Bug Additional Information...

### ISSUE THREAD

**Issue by usszulu2** [A3Wasteland/ArmA3_Wasteland.Altis#461]
This ticket is a direct response to ticket https://github.com/A3Wasteland/ArmA3_Wasteland.Altis/issues/305

I've decided to start a new ticket since I don't think the old one is getting any attention anymore.

Today when logging in, I noticed some anomalous lights at the vehicle store where vehicles normally spawn. Upon approaching it, I found that it appeared to be a vehicle that wasn't totally deleted. The headlights were present, but nothing else about the vehicle was. Before I started this recording, I did verify that it was acting as an invisible, solid object to vehicles, but didn't seem to react to people as solid. I then tried to delete the invisible vehicle with the admin delete vehicle command without success. I then bought a vehicle to test whether it would explode like I assumed it would, and it did as you can see in this video. 

So it appears that our bug is a not totally deleted vehicle. I am hoping this new information will help lead in a fix to this problem. 

Thanks!

Video:
https://youtu.be/qGOVJzPFnEw

**Comment by AgentRev** [comment]
The bug you referred is a different bug with the physics engine. What you experienced was probably a bug with the simulation manager, or a non-wasteland Arma glitch. The sim manager will probably be replaced by Bohemia's Dynamic Simulation in the near future.

**Comment by codexGW** [comment]
Yeah this could be what is affecting performance a bit because these ghost vehicles don't seem affected by client commands like deleteVehicle. So maybe if enough of these pile up and the fps fix wont work with them, causing lag

**Comment by codexGW** [comment]
It might not by a bad idea to add a warning and/or a check for driven vehicles in the last 5 minutes, then deleteVehicle 5m around the vehicle store spawns right before the server creates the new vehicle. Just in case.

**Comment by AgentRev** [comment]
The spawning script is supposed to detect nearby objects. If it didn't detect it then it's not close to the store from the script's perspective...

**Comment by usszulu2** [comment]
I guess my next question is since we do have these objects that are sometimes around vehicle spawn points and are invisible unless its night and the lights are on, is it possible or would it hurt to have the vehicles spawn a few feet off the ground to at least detect if they are in contact with an invisible object before getting into it and being blown up? Until BI implements the Dynamic Simulation at least?

**Comment by hobart666** [comment]
Yea, we've been getting a fancy anomaly since the last update, vehicles
that are destroyed still have the engine running and can hear it after
salvaging. I just figured it was BIS showing their prowess again. :P

**Comment by codexGW** [comment]
https://www.youtube.com/watch?v=A-tehMTk5og&feature=youtu.be

**Comment by WhatsUpDoc127** [comment]
You can easily just move the vehicle store spawn to a different spot on the map within Eden. Just move it over a bit. We've never had these issues on Altis nor Australia. Sometimes people places these thing too close to other objects and AgentRev is correct is appears an issue with the Simulation Manager. This reminds me of the Kuma tank that jumps all over the place, even today, whenever its gets near certain objects or heck it can be near nothing and it will jump all around.

**Comment by codexGW** [comment]
The spawn works fine when I tested it on my test servers and this obviously
happens on default spawns as well. This is a long-standing bug and it
usually happens under server load. For some reason if you shoot the vehicle
that will prevent it from exploding most of the time. There's a link to the
original issue in the first comment.

**Comment by mazer-rakham** [comment]
After update still exploding, bought a pawnee, unlocked got in, and as soon as I hit shift it exploded and said I suicided. Not to worried about refund, just wanna help the bug get fixed.

**Comment by AgentRev** [comment]
Dynamic Simulation was added in last week's update, I will investigate to see if it's a suitable replacement for my vehicle manager.

I just hope it will not delay the explosion of cached vehicles until they come within range, I remember having to make a workaround in my script, after witnessing exploding aircraft "ghosts" randomly popping out of the sky.

The ghost headlights may also have been caused by headless caching. If Dynamic Simulation meets my expectations, that won't be needed anymore.

It will unfortunately not fix the older vehicle store explosion bug, which I think is a physics bug in the engine.

**Comment by codexGW** [comment]
Sweet! Anything that might help performance is always time well spent in my opinion. Thank you.

**Comment by AgentRev** [comment]
I highly doubt it will make any noticeable difference in terms of performance.

**Comment by hobart666** [comment]
The most recent git performs phenomenally well, a relatively stable 50fps &
50 cps with 6 missions running.



On Mar 24, 2017 9:02 PM, "Agent Revolution" <notifications@github.com>
wrote:

I highly doubt it will make any noticeable difference in terms of
performance.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/A3Wasteland/ArmA3_Wasteland.Altis/issues/461#issuecomment-289176467>,
or mute the thread

**Comment by codexGW** [comment]
Why are you doing this Agent? Just to remove bloat, and bugs? Its always a bonus getting rid of another thing to maintain too. Definite bonus if it helps performance on servers with a lot of vehicles.

The performance is good in A3W as long as you don't go much over 40 players for much more than 2 hours, I have noticed.

**Comment by AgentRev** [comment]
Well, my vehicle manager does put a bit of a strain on the client script scheduler. Fortunately the scheduler is not tied to FPS, however replacing my manager with DynSim (which I presume is native C++) will free up the scheduler and make scripts a bit more responsive.

**Comment by ghost** [comment]
Too many Vehicles kill server performance that is for sure, if you let them get out of hand then it becomes a lag fest for clients. Looking forward to seeing this Rev

**Comment by WhatsUpDoc127** [comment]
We run the Australia A3W server. We use prebuilt bases, we do not allow building of bases as there is enough prebuilt bases. matt is correct too many vehicles will cause performance but I can tell you objects alone will cause performance issues. We have a huge blufor base I built and that alone without any vehicles inside causes lag. If I am in the base I get around 30fps but once away from the base I get 50-60+ fps easily. I still believe even with the 64bit executables that if BIS does not start utilizing the GPU more than what they are doing now it doesn't make a difference what Rev does. Whenever objects are added and performance is slower then that is on the game engine not A3W. Maybe just maybe BIS will get their heads out of their duffle bags some time in the near future and start using what other games are using, new and updated DirectX and GPU hardware.

For me I guess I need to look into to see if I can add prebuilt bases as part of the map instead of adding it to the editor, this will fix lag with adding objects in the editor itself.

**Comment by AgentRev** [comment]
If you build bases in Eden, you can use the Simple Object checkbox to make static objects a bit less CPU-intensive. This will freeze them in time and space.
